common/src/main/java/io/netty/util/internal/StringUtil.java

```diff
@@ -425,6 +426,78 @@ public static CharSequence unescapeCsv(CharSequence value) {
     }
 
     /**
+     * Unescapes the specified escaped CSV fields according to
+     * <a href="https://tools.ietf.org/html/rfc4180#section-2">RFC-4180</a>.
+     *
+     * @param value A string with multiple CSV escaped fields which will be unescaped according to
+     *              <a href="https://tools.ietf.org/html/rfc4180#section-2">RFC-4180</a>
+     * @return {@link List} the list of unescaped fields
+     */
+    public static List<CharSequence> unescapeCsvFields(CharSequence value) {
+        List<CharSequence> unescaped = new ArrayList<CharSequence>(2);
+        StringBuilder current = InternalThreadLocalMap.get().stringBuilder();
+        boolean quoted = false;
+        int last = value.length() - 1;
+        for (int i = 0; i <= last; i++) {
+            char c = value.charAt(i);
+            if (quoted) {
+                switch (c) {
+                    case DOUBLE_QUOTE:
+                        if (i == last) {
+                            // Add the last field and return
+                            unescaped.add(current.toString());
+                            return unescaped;
+                        }
+                        char next = value.charAt(i + 1);
```

Comment:
consider optimistically incrementing `i` here (`++i`) and decrement if found to be invalid `throw newInvalidEscapedCsvFieldException(value, i - 1);`